plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/StmtConverter.kt

```diff
@@ -65,4 +70,16 @@ class StmtConverter<out RTC : ResultTrackingContext>(
         get() = resultCtx.resultExp
 
     override fun capture(exp: Exp, expType: TypeEmbedding) = resultCtx.capture(exp, expType)
+
+    override fun inNewWhileBlock(action: (StmtConversionContext<ResultTrackingContext>) -> Unit) {
+        val freshIndex = newWhileIndex()
+        val continueLabel = Label(ContinueLabelName(freshIndex), listOf())
+        val breakLabel = Label(BreakLabelName(freshIndex), listOf())
```

Comment:
I think it would be good to expose these via the `WhileStack` interface, so you don't have to construct them over and over again.  This would then replace `whileIndex` in the interface.